### PR TITLE
Config cleanup

### PR DESCRIPTION
**Pull request by keradus**
Goal is to clean up the Config and it's usage

### CODE REVIEW

**Comment by SpacePossum** [comment]
thanks for picking this up, will be a great improvement : )

**Comment by SpacePossum** [comment]
Out of curiosity, do the attr. of the `Config` class need to be/stay `protected`? It feels like hard endpoints to maintain (BC) without a lot of gain.

**Comment by keradus** [comment]
Valid question.
I want to make a Config a DTO object without many deps, clean it up from unneeded props and then let us see.

**Comment by keradus** [comment]
https://github.com/keradus/PHP-CS-Fixer/pull/16 is merged into this PR

File: src/Console/ConfigurationResolver.php [hunk 1]
```diff
@@ -37,17 +45,17 @@
     const PATH_MODE_INTERSECTION = 'intersection';
 
     /**
-     * @var bool
+     * @var null|bool
```
Comment (SpacePossum) [inline src/Console/ConfigurationResolver.php]: can this really be `null` and what would it mean?

Comment (keradus) [inline src/Console/ConfigurationResolver.php]: Property is lazy loaded and should be accessed only via getter, which output is not nullable.

**Comment by keradus** [comment]
Thank you @SpacePossum.

File: src/Console/ConfigurationResolver.php (hunk 1 above)
Comment (SpacePossum) [inline src/Console/ConfigurationResolver.php]: :+1: I see now

**Pull request merged**